test-integration/operator-tests/src/test/resources/infinispans/client_validation.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: client-validation
+  annotations:
+    infinispan.org/monitoring: 'false'
+spec:
+  expose:
+    type: Route
+  replicas: 1
+  security:
+    endpointAuthentication: true
+    endpointEncryption:
+      type: Service
```

Comment:
In my opinion it would make more sense to use custom provided certificates for Infinispan server given custom certificates are used on the server side. I'm assuming that if the user configures custom certificates on client side it's more likely the user will use custom one also for the server. Opinion @ryanemerson ?